src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexNode.cs

```diff
@@ -310,18 +310,18 @@ internal RegexNode FinalOptimize()
                 // to implementations that don't support backtracking.
                 EliminateEndingBacktracking(rootNode.Child(0), DefaultMaxRecursionDepth);
 
-                // Optimization: unnecessary re-processing of atomic starting groups.
-                // If an expression is guaranteed to begin with a single-character infinite atomic group that isn't part of an alternation (in which case it
+                // Optimization: unnecessary re-processing of starting loops.
+                // If an expression is guaranteed to begin with a single-character unbounded loop that isn't part of an alternation (in which case it
```

Comment:
`If an expression is guaranteed to begin with a single-character`

I'm missing what part checks for a single character. Can you point it out to me?

Reply:
I wasn't sure how to phrase it better.  By "single character", I meant loops around a single character match, e.g. One/NotOne/Set... as opposed to loops around arbitrary expressions, e.g. Loop.  So we will optimize `\w*` but not `(\w\d)*`.